iib/web/models.py

```diff
@@ -1013,6 +1055,17 @@ def from_json(cls, kwargs, batch=None):
 
         request = cls(**request_kwargs)
         request.add_state('in_progress', 'The request was initiated')
+        db.session.add(request)
+        db.session.flush()
+
+        build_tags = []
+        if 'build_tags' in request_kwargs:
+            build_tags = request_kwargs.pop('build_tags')
```

Comment:
What about request_kwargs.pop('build_tags', []) instead of those 3 lines?
This logic can be done in Request._from_json() to avoid uplicates

Reply:
yeah, this can be reduced into one line